### PR TITLE
Improve keyboard navigation in tables

### PR DESCRIPTION
**Pull request by devongovett**
This includes several improvements to keyboard navigation behavior in tables:

* It's now possible to tab in and out of a table from outside the window (e.g. iframe, or browser chrome). I've replaced the previous strategy of preventing default on Tab key press and manually moving focus to the next/previous element from the table, with a different strategy, which moves focus to the first/last element *within* the table during the Tab key event and lets the browser move to the next/previous element outside the table (which could be outside the window/iframe).
* Tabbing also works properly from the column headers (ref was on the wrong element).
* No longer relies on detecting whether the tab key is pressed when focusing an element inside the table, which didn't work when tabbing in from outside an iframe or from the browser chrome (the keydown event wasn't fired inside). Now we do the inverse check using focus visible: if the user focuses with the mouse, then set the focused key, otherwise restore focus to the previously focused element.
* Removed the flicker when navigating to cells with focusable elements inside them where you'd see the cell itself focused first, and then the thing inside. Now it focuses the thing inside directly, rather than waiting for the next frame.
* Fixed keyboard navigation on menu buttons inside a table cell. Previously, ArrowDown/ArrowUp would open the menu, and there was no way to move to the cell above/below. Now they move to the cell above/below, except if you press the Alt key, in which case the menu is opened instead.

### CODE REVIEW

**Comment by snowystinger** [comment]
I can get into a state using the keyboard to navigate where i can't focus anything in the table by deleting a row. The collection I think is trying to use the last focused key it had

**Comment by devongovett** [comment]
For now I'm just resetting the focusedKey back to null if it no longer exists. Eventually we may want something more complex, like moving it to the previous/next row. Right now, since focus is in the menu, it doesn't really get restored back to the table anyway after the button that opened the menu unmounts. That's a bigger problem for another day.

File: packages/@react-stately/actiongroup/src/useActionGroupState.ts [hunk 1]
```diff
@@ -29,6 +29,13 @@ export function useActionGroupState<T extends object>(props: ActionGroupProps<T>
     return new TreeCollection(nodes);
   }, [builder, props, disabledKeys]);
 
+  // Reset focused key if that item is deleted from the collection.
```
Comment (snowystinger) [inline packages/@react-stately/actiongroup/src/useActionGroupState.ts:32]: should we make this a hook for collections? given that it was just copied into 4 files

Comment (devongovett) [reply packages/@react-stately/actiongroup/src/useActionGroupState.ts:32]: I actually tried that but it was a weird hook. Couldn't figure out what to call it!

Comment (dannify) [reply packages/@react-stately/actiongroup/src/useActionGroupState.ts:32]: I don't know if I would go for a whole hook just for this useEffect. We will probably have to refine the reset of focus in the near future so we can revisit it then.

**Pull request merged**